content/en/tracing/guide/add_span_md_and_graph_it.md

```diff
@@ -0,0 +1,287 @@
+kind: guide
+disable_toc: true
+further_reading:
+- link: "tracing/setup/"
+  tag: "Documentation"
+  text: "Learn how to setup APM tracing with your application"
+- link: "tracing/visualization/services_list/"
+  tag: "Documentation"
+  text: "Discover the list of services reporting to Datadog"
+- link: "tracing/visualization/resource"
+  tag: "Documentation"
+  text: "Dive into your resource performance and traces"
+- link: "tracing/visualization/trace"
+  tag: "Documentation"
+  text: "Understand how to read a Datadog Trace"
+---
+
+{{< img src="tracing/guide/add_span_md_and_graph_it/span_md_6.gif" alt="comparaison 2" responsive="true" style="width:90%;">}}
+
+Datadog APM allows you to customize your traces to include any additional information you might need to maintain observability into your business. In this example a customer ID will be added to traces allowing the customers that have the slowest performance to be identified. Customization of traces is based on tags that seamlessly integrate APM with the rest of Datadog and come in the form of key:value pairs of metadata that is added to spans. You can read more about tags and their use in Datadog in the [Tags Overview][1] page.
+
+## Instrument your code with custom metadata
+
+1. **Follow the example to get your code instrumented**.
+
+    Depending on the language you are you using, you’ll need to set the metadata to add to your spans differently.
+
+    **Note**: take note of the service and [resource names][2] you are working on, these will come in handy later. In this example the service is the Ruby server `web-store` and the resource (endpoint) is `ShoppingCartController#checkout`.
+
+{{< tabs >}}
+{{% tab "Java" %}}
+
+The Datadog UI uses tags to set span level metadata. A full list of these metadata can be found in the [Datadog][1] and [OpenTracing][2] APIs.
+
+Custom metadata may be set for auto-instrumentation by grabbing the active span out of the global tracer and setting a tag with `setTag`.
+
+```java
+import io.opentracing.Tracer;
+import io.opentracing.util.GlobalTracer;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet
+class ServletImpl extends AbstractHttpServlet {
+  @Override
+  void doGet(HttpServletRequest req, HttpServletResponse resp) {
+    final Span span = GlobalTracer.get().activeSpan();
+    if (span != null) {
+      span.setTag("customer.id", 12345);
+      span.setTag("http.url", "/login");
+    }
+    // servlet impl
+  }
+}
+```
+
+[1]: https://github.com/DataDog/dd-trace-java/blob/master/dd-trace-api/src/main/java/datadog/trace/api/DDTags.java
+[2]: https://github.com/opentracing/opentracing-java/blob/master/opentracing-api/src/main/java/io/opentracing/tag/Tags.java
+{{% /tab %}}
+{{% tab "Python" %}}
+
+Add metadata directly to a span by calling `set_tag`. For example, with the following route handler:
+
+```python
```

Comment:
Do we really need that section for most of the languages? Isn't enough showing only the second snippet about "how to add custom tags"?

In Java we have:
> Custom metadata may be set for auto-instrumentation by grabbing the active span out of the global tracer and setting a tag with `setTag`

That I think is cool because it shows:
- you have auto-instrumentation enabled
- you enrich auto-instrumentation without doing anything more
- having two code-blocks could be misleading. It may let people think you need to do "a lot of things" to achieve that, while if we focus only on a) retrieving the span b) adding the tag, it will be straightforward

This is just a thought, in general I prefer to keep at minimum the code we show, unless it's really needed to understand the topic.

Reply:
I completely agree and think that for the most part we can remove one of the blocks for every language. I'm just not confident I can do this myself. Would you be ok assigning this to folks on your team to just make the modifications needed? 

Purpose would be for each language tab to have the minimal code needed to add "customer_id" to a span and exemplify (as you say) that auto-instrumentation is great and and enriching it is easy.

Reply:
Agreed this should be as simple as possible and only the minimal code.

Also they should all read customer_id: 22495 to match the screenshot - and we should show that more quickly in this doc to see the immediate impact from code -> UI.

Reply:
I think when possible they should have a parameterized customer ID, not look like we're hard coding a number, as Manu mentioned in the note to @delner below...

Reply:
We can add a comment before the parameter to say the value they will find in the screenshot later. Let me work on this.